### DISCUSSION TITLE
Is it possible to prevent text reflow?

### DISCUSSION THREAD

**Discussion by ramiromagno** [quarto-dev/quarto-cli#6272] [Q&A] [answered]
### Description

When exporting to Latex (at least), is it possible to prevent the reflow of text?

I am trying to ensure that a latex template generated with Quarto remains as identical as possible to the journal's provided template but it really does not help not being able to control how lines are broken...

![snap](https://github.com/quarto-dev/quarto-cli/assets/19862384/0d6377bf-55b7-4f20-bccb-587e01586c84)

**Comment by mcanouil** [comment]
If you are using a template (LaTeX?) as base, why not use that directly?
See <https://quarto.org/docs/journals/formats.html>.

**Reply by ramiromagno** [reply]
I am not using a Quarto Latex template, I am trying to develop one: https://github.com/ramiromagno/wiley-njd/. What I have as starting point is the class (.cls), and a bunch of .sty from the publisher, along with a .tex example file --- this file is the publisher's latex template, and I am trying to have one in Quarto.

**Reply by mcanouil** [reply]
Is it not defined in the class?

Anyway, take a look at Quarto partials (<https://quarto.org/docs/journals/templates.html#template-partials>), for the default LaTeX template in Quarto.

For LaTeX question, you should consider looking on dedicated platforms as most LaTeX questions have already been asked and answered on the Internet.

**Reply by ramiromagno** [reply]
Thank you for your answer. I do have quite some of that done already here: https://github.com/ramiromagno/wiley-njd/tree/main/_extensions/wiley-njd. And I do use profusely other platforms for LaTeX questions, for many years now. But this is not a LaTeX question per se.

**Comment by baptiste** [comment]
This sounds like line wrapping, which can be influenced by the text editor itself – e.g. [for the visual editor here](https://quarto.org/docs/visual-editor/markdown.html#line-wrapping)

So I guess the question is what is your source document and what is being done to it? (your screenshot shows two latex files, so I'm not sure where quarto enters the picture)

**Reply by mcanouil** [reply]
Exactly, before the last comment I thought it was about the output (LaTeX) but the last comment it seems to be about the line wrapping which should have no impact on the output (and little to do with Quarto).

**Reply by rgaiacs** [reply]
@ramiromagno By default Pandoc will hard coded wrap lines. Because of this, diff between two `.tex` files will be hard as a small change will be reflect in a couple of lines because of the hard wrap lines. My suggestion is to use Pandoc's `--wrap=none`.

# Minimal Working Example

`quarto render mwe.qmd --to latex` will convert

```
---
title: MWE
author: Raniere Silva
---

Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Mattis molestie a iaculis at erat. Posuere ac ut consequat semper viverra.
```

into

```
\title{MWE}
\author{Raniere Silva}
\date{}

\begin{document}
\maketitle
Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod
tempor incididunt ut labore et dolore magna aliqua. Mattis molestie a
iaculis at erat. Posuere ac ut consequat semper viverra.



\end{document}
```

Pay attention to the hard coded wrap lines.

`quarto render mwe.qmd --to latex --wrap=none` will produce

```
\title{MWE}
\author{Raniere Silva}
\date{}

\begin{document}
\maketitle
Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Mattis molestie a iaculis at erat. Posuere ac ut consequat semper viverra.



\end{document}
```

Pay attention that the paragraph is kept in a single line.

You can include Pandoc's `--wrap=none` in Quarto's front matter.

**Reply by ramiromagno** [reply, marked as answer]
@rgaiacs Thank you for understanding my issue here. Your solution was 99% what I needed to know. Then I found this issue: https://github.com/jgm/pandoc/issues/2617, and the two pieces of information led me in the right direction!

So, now, for others that might be reading this discussion and that also want to go from qmd -> tex and that do not want quarto/pandoc reflowing the text, use the option:

```yaml
wrap: preserve
```
And it works in my `_extension.yml` too, which is great!

https://github.com/ramiromagno/wiley-njd/blob/25c9d234d852e5f0459c9eeaafb9dc7119a81c1b/_extensions/wiley-njd/_extension.yml#L30

**Reply by ramiromagno** [reply]
For others reading this, please note that this `wrap` option is not the same editor's option as documented here: https://quarto.org/docs/visual-editor/markdown.html#line-wrapping. This last one is under `editor` / (e.g. `markdown`) / `wrap`. The one presented as solution is in `_extension.yml`, under `contributes` / `formats` / `pdf` / `wrap` --- to my knowledge this is not documented in Quarto.
